src/snapshot.zig

```diff
@@ -217,14 +217,126 @@ const Section = union(enum) {
     };
 };
 
+/// The type of node to parse
+pub const NodeType = enum {
+    file,
+    header,
+    expr,
+    statement,
+
+    const HEADER = "header";
+    const EXPR = "expr";
+    const STMT = "statement";
+    const FILE = "file";
+
+    fn fromString(ty: []const u8) Error!NodeType {
+        if (std.mem.eql(u8, ty, HEADER)) return .header;
+        if (std.mem.eql(u8, ty, EXPR)) return .expr;
+        if (std.mem.eql(u8, ty, STMT)) return .statement;
+        if (std.mem.eql(u8, ty, FILE)) return .file;
+        return Error.InvalidNodeType;
+    }
+
+    fn toString(self: NodeType) []const u8 {
+        return switch (self) {
+            .file => "file",
+            .header => "header",
+            .expr => "expr",
+            .statement => "statement",
+        };
+    }
+};
+
+const Meta = struct {
+    description: []const u8,
+    node_type: NodeType,
+
+    const DESC_START = "description=";
+    const TYPE_START = "type=";
+
+    fn fromString(text: []const u8) Error!Meta {
+        var lines = std.mem.splitScalar(u8, text, '\n');
+        var desc: []const u8 = "";
+        var node_type: NodeType = .file;
+        while (true) {
+            var line = lines.next() orelse break;
+            if (std.mem.startsWith(u8, line, DESC_START)) {
+                desc = line[12..];
+            } else if (std.mem.startsWith(u8, line, TYPE_START)) {
+                const ty = line[5..];
```

Comment:
Instead of using `12` and `5` which feel like magic numbers, can we use `DESC_START.len` adn `TYPE_START.len`?

Reply:
Definitely

Reply:
DONE